src/__swaps__/screens/Swap/components/CoinRow.tsx

```diff
@@ -11,7 +11,104 @@ import Animated from 'react-native-reanimated';
 import { StyleSheet } from 'react-native';
 import { useSwapContext } from '@/__swaps__/screens/Swap/providers/swap-provider';
 import { SwapCoinIcon } from './SwapCoinIcon';
-import { ethereumUtils } from '@/utils';
+import { ethereumUtils, haptics, showActionSheetWithOptions } from '@/utils';
+import { ContextMenuButton } from 'react-native-ios-context-menu';
+import { IS_ANDROID } from '@/env';
+import { startCase } from 'lodash';
+import { setClipboard } from '@/hooks/useClipboard';
+import { RainbowNetworks } from '@/networks';
+import * as i18n from '@/languages';
+
+const InfoButton = ({ address, chainId }: { address: string; chainId: ChainId }) => {
+  const network = RainbowNetworks.find(network => network.id === chainId)?.value;
+
+  const handleCopy = useCallback(() => {
+    haptics.selection();
+    setClipboard(address);
+  }, [address]);
+
+  const options = {
+    copy: {
+      title: i18n.t(i18n.l.exchange.coin_row.copy_contract_address),
+      action: handleCopy,
+    },
+    ...(network
+      ? {
+          blockExplorer: {
+            title: i18n.t(i18n.l.exchange.coin_row.view_on, { blockExplorerName: startCase(ethereumUtils.getBlockExplorer(network)) }),
```

Comment:
i don't think you're consuming the `blockExplorerName` in the i18n entry.

Reply:
this is correct right?

<img width="476" alt="Screenshot 2024-05-31 at 9 14 28 AM" src="https://github.com/rainbow-me/rainbow/assets/15272675/401ba73b-379f-4385-8efb-466a2606e4ed">

Reply:
oh weird the git diff didn't show that, my bad